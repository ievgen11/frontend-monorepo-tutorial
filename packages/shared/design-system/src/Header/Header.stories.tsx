import * as React from "react";
import { withKnobs, text } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";

import { Header, ClientHeader } from "./index";

export default {
  title: "Header",
  component: Header,
  decorators: [withKnobs],
};

export const Default = () => (
  <Header>{text("Text", "Track Your Package")}</Header>
);

export const ClientLogo = () => (
  <ClientHeader onLogoClick={action("onLogoClick")}>
    {text("Text", "Client Logo")}
  </ClientHeader>
);
