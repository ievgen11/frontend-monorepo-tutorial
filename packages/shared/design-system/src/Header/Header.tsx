import React from "react";
import styled from "styled-components";

const Wrapper = styled.header`
  width: 100%;
  height: auto;
  background-color: transparent;

  @media only screen and (min-width: 768px) {
    height: 150px;
  }
`;

const Container = styled.div`
  display: flex;
  padding: 20px 24px 30px 24px;
  position: unset;
  background-color: #f0f0f0;
  box-sizing: border-box;
  box-shadow: 0 0 16px 32px #f0f0f0;
  z-index: 1;
  flex-direction: column;
  align-items: flex-start;

  @media only screen and (min-width: 768px) {
    width: 100%;
    margin: auto;
    padding: 64px 80px 30px 80px;
    flex-direction: row;
    align-items: center;
    max-width: 1200px;
  }
`;

const LogoContainer = styled.div`
  flex-grow: 1;
  margin-bottom: 12px;
  margin-right: 24px;

  @media only screen and (min-width: 768px) {
    margin-bottom: 0;
  }
`;

const Logo = styled.img`
  height: 25px;
  width: auto;
  cursor: pointer;

  &:hover {
    opacity: 0.6;
  }
`;

import clientLogo from "./images/client_logo.png";
import adminLogo from "./images/admin_logo.png";

export interface HeaderProps {
  children?: React.ReactNode;
  logo?: string;
  onLogoClick?: () => void;
}

export const Header: React.FC<HeaderProps> = ({
  children,
  logo,
  onLogoClick,
}): JSX.Element => (
  <Wrapper>
    <Container>
      {logo ? (
        <LogoContainer>
          <Logo onClick={onLogoClick} src={logo} />
        </LogoContainer>
      ) : null}
      {children}
    </Container>
  </Wrapper>
);

export const ClientHeader: React.FC<HeaderProps> = (props): JSX.Element => (
  <Header logo={clientLogo} {...props} />
);

export const AdminHeader: React.FC<HeaderProps> = (props): JSX.Element => (
  <Header logo={adminLogo} {...props} />
);
