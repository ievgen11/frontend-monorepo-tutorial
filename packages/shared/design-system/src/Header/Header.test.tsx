import * as React from "react";
import * as renderer from "react-test-renderer";

import { Header } from "./Header";

describe("<Header />", () => {
  test("Snapshot is intact", () => {
    const tree = renderer.create(<Header>Test</Header>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
