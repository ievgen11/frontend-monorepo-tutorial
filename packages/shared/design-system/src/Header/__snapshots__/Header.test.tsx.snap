// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Snapshot is intact 1`] = `
<header
  className="sc-bdfBwQ dfRjnf"
>
  <div
    className="sc-gsTCUz hzLaqM"
  >
    Test
  </div>
</header>
`;
